### DISCUSSION TITLE
Fooocus-Control：add controlnet to fooocus

### DISCUSSION THREAD

**Discussion by fenneishi** [lllyasviel/Fooocus#786] [Show and tell]
Hi,

I'm excited to share Fooocus-Control. 
Fooocus-Control is a ⭐free⭐ image generating software (based on [Fooocus](https://github.com/lllyasviel/Fooocus) , [ControlNet](https://github.com/lllyasviel/ControlNet-v1-1-nightly) ,👉[SDXL](https://github.com/Stability-AI/generative-models) , [IP-Adapter](https://ip-adapter.github.io/) , etc.).
Fooocus-Control adds more control to the original Fooocus software.
---
github: https://github.com/fenneishi/Fooocus-ControlNet-SDXL
---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/9a45d616-a9d3-428f-b8c4-daf819a3eaaf)
---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/0921ec61-d459-460e-b392-94e4af8b0932)
---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/5253ee07-0fdc-446f-a8c5-37aab7212fa4)

---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/ce7d4cab-9103-49e6-b8fc-91d70633a833)
---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/f3f9a20d-8dca-40a7-bab5-843138393730)
---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/102836f7-2d20-4918-a268-fd757909c61b)

---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/9a884618-77a7-4076-94d0-60d8cc73c12f)
---
![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/f550e1eb-1e01-4327-9ca9-90ae886dffed)

---
### 💡 FooocusControl keeps it simple

- Fooocus is an excellent SDXL-based software, which provides excellent generation effects based on the simplicity of
  liking midjourney, while being free as stable diffusiond.
- FooocusControl inherits the core design concepts of fooocus, in order to minimize the learning threshold,
  FooocusControl has the same UI interface as fooocus (only in the Input Image/Image Prompt/advance to add more
  options).
- FooocusControl does all the complicated stuff behind the scenes, such as model downloading, loading, registration,
  image preprocessing, etc. Users don't need to bother with any of this at all, you just need to check the desired image
  control method.

### 💡 FooocusControl facilitates secondary development

- FooocusControl simplifies the way fooocus integrates with controlnet by simply defining pre-processing and adding
  configuration files.
- If you are a developer with your own unique controlnet model , with FooocusControl , you can easily integrate it into
  fooocus .
- In addition to controlnet, FooocusControl plans to continue to integrate ip-adapter and other models to further
  provide users with more control methods.

### 💡 FooocusControl pursues the out-of-the-box use of software

- Free software usually encounters a lot of installation and use of the problem, such as network problems caused by the
  model file that can not be downloaded and updated, a variety of headaches gpu driver, plug-ins lack of dependent
  libraries and other issues. These are very annoying invisible thresholds. fooocusControl is committed to solving these
  problems.
- FooocusControl has no external dependencies, all the dependencies are built into the software.
- FooocusControl provides both light and massive download methods, the latter will download all the models built into
  the software, which is very friendly to users who need to run offline, or those who have a bad network.
- Outside the windows platform, FooocusControl will try to use docker and other technologies to further simplify the
  user installation, to avoid various gpu and dependency problems to the greatest extent possible.
- FooocusControl will add multiple download sources for each model (coming soon) to prevent models from being downloaded
  or updated due to network problems.
---

# [ Free Install Fooocus-Control](#Install)

# <center>🍇show case🍇</center>

# Sketch

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/b7864f73-c257-4433-96e9-54a4530d68dc)

---

# ImagePrompt

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/bbb6e197-50da-4ac2-a2fe-e67e20c170b8)

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/f257c641-aa05-4f7d-975f-db12b419a571)

---

# Canny

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/49b25dfe-3266-4383-a0f5-9c88b2ab8a03)

---

# Depth

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/b2a7a60f-5541-4c5c-a0cf-a6300bc66653)

---

# Pose

:face can control pose

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/27a58c7c-49ed-4262-a27b-296aa858c413)

---

# Recolor

![image](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/146184643/15e2d31a-e410-477f-b7e6-d8944771e01e)

---

**Comment by barepixels** [comment]
I tested Depth.  Works great.  Here is my result 
(NSFW) https://i.imgur.com/QFfParx.jpg (NSFW)

You should have Issues and Discussion activated on your GitHub so people can leave feedback

**Comment by fenneishi** [comment]
happy to help you.

**Comment by fenneishi** [comment]
The Issues and Discussions have been activated.

**Reply by adafym** [reply]
Yeah, need this.

**Comment by adafym** [comment]
Well, I have tested depth and pose, it works great!!!

**Comment by lllyasviel** [comment]
Thanks for the work.
This repo has been added to https://github.com/lllyasviel/Fooocus/discussions/699 and some related fork promotions are added to several places.
Feel free to see also related policy and more details in the above link.

**Reply by fenneishi** [reply]
Thank you for adding this repo to #699 and promoting the forks. I've reviewed the related policy in the link. Looking forward to contributing more to the community!

**Comment by PeiPeiWow** [comment]
Great!!!

**Comment by kantsche** [comment]
it would be good to have a start at and stop at

**Reply by fenneishi** [reply]
![image](https://github.com/lllyasviel/Fooocus/assets/33345538/85327155-dc61-4da0-8a01-3c8df4d78e98)
Is that what you mean？

**Reply by kantsche** [reply]
yes

**Reply by fenneishi** [reply]
Some experiments in the paper have shown that "**_Obviously, the later the iteration, the less the guiding effect._**". 
Therefore, I followed the UI design of Fooocus, which includes only 'stop at' functionality, without 'start at'.
![image](https://github.com/lllyasviel/Fooocus/assets/33345538/8fc50db7-b876-4d80-b917-b5b3def7dff6)
ref:[T2I-Adapter: Learning Adapters to Dig out More Controllable Ability for Text-to-Image Diffusion Models](https://arxiv.org/abs/2302.08453)

**Reply by kantsche** [reply]
for image prompts its better, because if you guide it from the beginning the image gonna clash with the text prompt

**Reply by fenneishi** [reply]
Can you give me some image evidence instance?If it's not too much trouble.

**Reply by fenneishi** [reply]
Or some research？

**Reply by kantsche** [reply]
This image used for ip adapter
![image](https://github.com/lllyasviel/Fooocus/assets/5844962/db805252-9dee-4d18-897e-ca19e324f154)
with the prompt "portait dog painting"

stop at 1.0
![image](https://github.com/lllyasviel/Fooocus/assets/5844962/137589e6-7ef2-49a4-9f4b-cda8769ce8c0)

stop at 0.5
![00008-3627918862](https://github.com/lllyasviel/Fooocus/assets/5844962/8d923aaf-2fa1-436c-9585-6ee297bf6eac)

start at 0.5
![image](https://github.com/lllyasviel/Fooocus/assets/5844962/04c7f0db-3bd4-42a4-822d-8584ee099cd5)

start at 0.6
![image](https://github.com/lllyasviel/Fooocus/assets/5844962/915a64af-01d9-4ce6-9dc6-071f891c347e)

start at 0.4 stop at 0.6 (but with higher weight, foocus can only go to 1.0)
![image](https://github.com/lllyasviel/Fooocus/assets/5844962/fe630b85-b0cf-4f3f-a296-7271ee9ac579)

**Reply by fenneishi** [reply]
Thank you very much for your feedback，I will actively consider this feature.

My thinking is: 

**_Essentially,this is a bug(or flaw) at the algorithmic layer instead of software layer_**

So, the complete  solution is to resolve the conflicts between the various prompts ,This paper may point to an idea for the solution：
[《Mix-of-Show: Decentralized Low-Rank Adaptation for Multi-Concept Customization of Diffusion Models》](https://showlab.github.io/Mix-of-Show/)

As a temporary solution, I might prefer one of the following：
* Introduce a choice button in image prompt: **content** or **style**. content with default "start at=0.0" and style with default 'start at = 0.5'.
* Add a new control named : "**image prompt(style)**" with the default 'start at=0.5' and the ‘start at’ can be config by user.

thanks again!

**Reply by fenneishi** [reply]
developing  🤘✌️
![微信图片_20231030013938](https://github.com/lllyasviel/Fooocus/assets/33345538/c67b3ee7-7475-4283-8fe7-fdfee8475602)

**Comment by Yaozong-Lee** [comment]
is it different from fooocus-MRE?

**Comment by RompinDonkey** [comment]
Hello.  Noobie to AI gen software here (apologies if my question is silly)  Interested in adding this to Fooocus which I have loved so far. I am limited in software I can use due to my PC specs. That's the beauty of Fooocus, my 1060 6gb card and i5 7600 is handling things perfectly, gens coming out nice and fast with pleasing results etc.  Would adding FooocusControl change that? Would it still run as smoothly and efficiently as without FC?

Edit: Or have I misunderstood - IS FC a separate software?  I originally assumed it was some kind of extension or add-on to the original Fooocus. Apologies if I misunderstood.

 Thanks.

**Comment by YourKoach** [comment]
I already have the original Fooocus installed. Do I need to uninstall it before I can install Fooocus-Control?

**Reply by mashb1t** [reply]
Fooocus-Control is a separate project and can simply be downloaded  and installed in parallel by using a different folder name.
